backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java

```diff
@@ -1621,21 +1637,25 @@ private Map<String, Map<String, CodeVersion>> getExperimentRunCodeVersionMap(
             + " LEFT JOIN PathDatasetComponentBlobEntity pdcb ON ncb.path_dataset_blob_hash = pdcb.id.path_dataset_blob_id "
             + " WHERE vme.versioning_blob_type = :versioningBlobType AND vme.experimentRunEntity.id IN (:expRunIds) ";
 
+    Set<String> accessibleRepositoryIds;
     if (mdbConfig.isPopulateConnectionsBasedOnPrivileges()) {
-      if (selfAllowedRepositoryIds == null || selfAllowedRepositoryIds.isEmpty()) {
+      boolean allowedAllResources =
+          RoleServiceUtils.checkAllResourceAllowed(selfAllowedRepositories);
+      if (allowedAllResources) {
         return new HashMap<>();
       } else {
+        accessibleRepositoryIds = RoleServiceUtils.getResourceIds(selfAllowedRepositories);
         queryBuilder = queryBuilder + " AND vme.repository_id IN (:repoIds)";
       }
+    } else {
+      accessibleRepositoryIds = null;
```

Comment:
same comment here. null seems bad vs. an empty Set, and you can assign this value in the declaration, rather than in an `else` block.

Reply:
replaced it with empty set and moved to initializer